src/main/java/core/basesyntax/FieldXY.java

```diff
@@ -13,7 +13,27 @@
 public class FieldXY {
 
     public void moveRobot(Robot robot, int toX, int toY) {
-        //your code
+        boolean moveToLeft = false;
+        if (robot.getX() > toX) {
+            moveToLeft = true;
+        }
+
+        if (robot.getX() != toX) {
+            while (robot.getDirection() != (moveToLeft ? Direction.LEFT : Direction.RIGHT)) {
+                robot.turnLeft();
+            }
+            while (robot.getX() != toX) {
+                robot.stepForward();
+            }
+        }
+
+        if (robot.getY() != toY) {
+            while (robot.getDirection() != ((robot.getY() > toY) ? Direction.DOWN : Direction.UP)) {
```

Comment:
with this approach you code is not consistent: 
with x you create separate variable to hold boolean value(movetoLeft) and with y you don't do that. So it's a bit confusing and it's better to avoid those situations 
You can make you check more simple: 
Direction directionToX = (robot.getX() < toX) ? Direction.RIGHT : Direction.LEFT;

Reply:
done